src/library_syscall.js

```diff
@@ -137,10 +137,14 @@ var SyscallsLibrary = {
       }
       return 0;
     },
-    doDup: function(path, flags, suggestFD) {
+    doDup: function(stream, suggestFD) {
       var suggest = FS.getStream(suggestFD);
       if (suggest) FS.close(suggest);
-      return FS.open(path, flags, 0, suggestFD, suggestFD).fd;
+      if(stream.node.pipe){
```

Comment:
space after `if` and before `{`

Reply:
To be clear, I assume that there isn't an automatic formatter that won't choke on the `#ifdef`'s and stuff?

Reply:
I think `clang-format` can operate on just a single line or range of lines in your editor.. although I don't use it myself for JS much.